browser/registry_test.go

```diff
@@ -33,70 +33,110 @@ func TestPidRegistry(t *testing.T) {
-			},
+			name:           "multiple WS URL",
 			expIsRemote:    true,
 			expValidWSURLs: []string{"WS_URL_1", "WS_URL_2", "WS_URL_3"},
+			envVarName:     "K6_BROWSER_WS_URL",
+			envVarValue:    "WS_URL_1,WS_URL_2,WS_URL_3",
 		},
 		{
-			name: "ending comma is handled",
-			envLookup: func(key string) (string, bool) {
-				return "WS_URL_1,WS_URL_2,", true
-			},
+			name:           "ending comma is handled",
 			expIsRemote:    true,
 			expValidWSURLs: []string{"WS_URL_1", "WS_URL_2"},
+			envVarName:     "K6_BROWSER_WS_URL",
+			envVarValue:    "WS_URL_1,WS_URL_2,",
 		},
 		{
-			name: "void string does not panic",
-			envLookup: func(key string) (string, bool) {
-				return "", true
-			},
+			name:           "void string does not panic",
 			expIsRemote:    true,
 			expValidWSURLs: []string{""},
+			envVarName:     "K6_BROWSER_WS_URL",
+			envVarValue:    "",
 		},
 		{
-			name: "comma does not panic",
-			envLookup: func(key string) (string, bool) {
-				return ",", true
-			},
+			name:           "comma does not panic",
 			expIsRemote:    true,
 			expValidWSURLs: []string{""},
+			envVarName:     "K6_BROWSER_WS_URL",
+			envVarValue:    ",",
+		},
+		{
+			name:           "read a single scenario with a single ws url",
+			expIsRemote:    true,
+			expValidWSURLs: []string{"WS_URL_1"},
+			envVarName:     "K6_INSTANCE_SCENARIOS",
+			envVarValue:    `[{"id": "one","browsers": [{ "handle": "WS_URL_1" }]}]`,
+		},
+		{
+			name:           "read a single scenario with a two ws urls",
+			expIsRemote:    true,
+			expValidWSURLs: []string{"WS_URL_1", "WS_URL_2"},
+			envVarName:     "K6_INSTANCE_SCENARIOS",
+			envVarValue:    `[{"id": "one","browsers": [{"handle": "WS_URL_1"}, {"handle": "WS_URL_2"}]}]`,
+		},
+		{
+			name:           "read two scenarios with multiple ws urls",
+			expIsRemote:    true,
+			expValidWSURLs: []string{"WS_URL_1", "WS_URL_2", "WS_URL_3", "WS_URL_4"},
+			envVarName:     "K6_INSTANCE_SCENARIOS",
+			envVarValue: `[{"id": "one","browsers": [{"handle": "WS_URL_1"}, {"handle": "WS_URL_2"}]},
+			{"id": "two","browsers": [{"handle": "WS_URL_3"}, {"handle": "WS_URL_4"}]}]`,
+		},
+		{
+			name:           "read scenarios without any ws urls",
+			expIsRemote:    false,
+			expValidWSURLs: []string{""},
+			envVarName:     "K6_INSTANCE_SCENARIOS",
+			envVarValue:    `[{"id": "one","browsers": [{}]}]`,
+		},
+		{
+			name:           "read scenarios without any browser objects",
+			expIsRemote:    false,
+			expValidWSURLs: []string{""},
+			envVarName:     "K6_INSTANCE_SCENARIOS",
+			envVarValue:    `[{"id": "one"}]`,
+		},
+		{
+			name:        "read empty scenarios",
+			expErr:      errors.New("parsing K6_INSTANCE_SCENARIOS: unexpected end of JSON input"),
+			envVarName:  "K6_INSTANCE_SCENARIOS",
+			envVarValue: ``,
 		},
 	}
 
 	for _, tc := range testCases {
-		tc := tc
-
 		t.Run(tc.name, func(t *testing.T) {
-			t.Parallel()
+			t.Setenv(tc.envVarName, tc.envVarValue)
```

Comment:
@ankur22, thanks for making the latest changes about `panic`, LGTM!

One last suggestion. Haven't we defined the `env.Lookup` func also for testing? 🤔 WDYT about reenabling `t.Parallel` and passing a fake environment value provider function here?

Reply:
I made a trade off here for simplicity over speed. Since the `newRemoteRegistry` now looks for two env vars, the `envLookup` in the tests would also need to be able to handle both env vars, so the simple _"single WS URL"_ would have an `envLookup` method:

```
func(key string) (string, bool) {
  if key != "K6_BROWSER_WS_URL" {
	  return "", false
  }
  return "WS_URL", true
}
```

The speed gain by parallelising this specific unit test doesn't seem worth it.

If there's a strong opinion though i'll refactor the test in another PR, if that's ok?

Reply:
I understand the motivation, but I'm not sure a simple if case adds much complexity 🤔 I'm fine either way.

Reply:
Thanks @inancgumus. In that case I'll add a note to refactor this test in [#822](https://github.com/grafana/xk6-browser/issues/822) and maybe we can tackle this in the next couple of weeks 🙂

Reply:
:D Sure.

Reply:
Fixed [here](https://github.com/grafana/xk6-browser/pull/918/commits/ad81e72a044af49b63bed8918f946966b8740000).